controllers/openstackcluster_controller.go

```diff
@@ -297,8 +306,11 @@ func reconcileNormal(scope scope.Scope, cluster *clusterv1.Cluster, openStackClu
 	}
 
 	result, err := reconcileBastion(scope, cluster, openStackCluster)
-	if err != nil || !reflect.DeepEqual(result, reconcile.Result{}) {
-		return result, err
+	if err != nil {
+		return reconcile.Result{}, err
```

Comment:
here I can also comment that in reconcileBastion we might return a result with Requeuing.

I'm honestly not sure we want that but I'm happy to do it.